Scribble support for new acmart.cls.

dvanhorn opened the pull request.

bennn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+
+@deftogether[(
+@defproc[(acmJournal [journal pre-content?] ...) content?]
+@defproc[(acmConference [name pre-content?] [date pre-content?] [venue pre-content?]) content?]
```
no `short-name` optional argument?

dvanhorn replied:
What is the idiomatic way of supporting optional arguments in LaTeX commands?  Best I could come up with was manually define two latex wrapper commands.

bennn replied:
Good question, I don't know either.

dvanhorn replied:
I added the optional URL argument for `acmBadgeL` and `acmBadgeR` and can do the same for these (but I'll first see if there's a better way to do this).

bennn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+@defproc[(acmArticle [content pre-content?] ...) content?]
+@defproc[(acmYear [content pre-content?] ...) content?]
+@defproc[(acmMonth [content pre-content?] ...) content?]
+@defproc[(acmArticleSeq [content pre-content?] ...) content?]
```
For `acmVolume acmNumber acmArticle acmYear acmMonth acmArticleSeq`, I think `integer?` is the right contract & they should just take 1 argument.

(Can't think of why someone would want to pass something that doesn't evaluate to an integer)

dvanhorn replied:
It's easy to write strings in scribble and doesn't require remembering the right type to use.  This interface is more consistent with the latex interface _and_ the existing `sigplan` library.

bennn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+@defproc[(acmYear [content pre-content?] ...) content?]
+@defproc[(acmMonth [content pre-content?] ...) content?]
+@defproc[(acmArticleSeq [content pre-content?] ...) content?]
+@defproc[(acmPrice [content pre-content?] ...) content?]
```
Same, I think the domain should be `flonum?`

dvanhorn replied:
Same response.

bennn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+@defproc[(acmArticleSeq [content pre-content?] ...) content?]
+@defproc[(acmPrice [content pre-content?] ...) content?]
+@defproc[(acmISBN [content pre-content?] ...) content?]
+@defproc[(acmDOI [content pre-content?] ...) content?]
```
For `acmISBN` and `acmDOI` I think the domain should be `pre-content?`

dvanhorn replied:
Tried to be consistent with `sigplan`.

bennn replied:
Ah! that explains a lot. (I had never read the `scribble/sigplan` docs.)

bennn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+@defproc[(acmISBN [content pre-content?] ...) content?]
+@defproc[(acmDOI [content pre-content?] ...) content?]
+@defproc[(acmBadgeL [content pre-content?] ...) content?]
+@defproc[(acmBadgeR [content pre-content?] ...) content?]
```
For `acmBadgeL` and `acmBadgeR`, I think the domain should be `string? path-string?`
(or at least, the same as `hyperlink`'s domain)

dvanhorn replied:
I've switched it to `string?` (and will add the optional argument when I know how to).

bennn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+)]{
+
+Declares information that is collected for each author.  These commands should
+ only be used within an @racket[affiliation] command.}
```
Add an example?

dvanhorn replied:
OK.

bennn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+word, separate phrases by commas, and do not include ``and'' before
+the last one. Keywords should be noun phrases, not adjectives.}
+
+@defproc[(startpage [content pre-content?] ...) content?]{
```
Should it be `startPage`?
(To match the macro I see in the acmart v1.28 docs)

bennn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+
+Sets the date the paper was received.}
+
+@defproc[(citestyle [content pre-content?] ...) content?]{
```
Does this need the `...` ?

bennn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+
+@defproc[(acks [content pre-content?] ...) block?]{
+
+Creates an acknowledgments section.}
```
unlabeled? or does it do what acmart's `ack` does?

dvanhorn replied:
Does what `ack` does.

dvanhorn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+
+@defmodulelang[scribble/acmart]{The @racketmodname[scribble/acmart]
+language is like @racketmodname[scribble/base], but configured with
+Latex style defaults to use the @filepath{acmart.cls} class
```
Just like `sigplan`, but I'll change both.

dvanhorn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+@defacmartid[screen]
+@defacmartid[natbib]
+@defacmartid[anonymous]
+@defacmartid[authorversion]
```
OK.

dvanhorn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+
+@defproc[(abstract [pre-content pre-content?] ...) block?]{
+
+Generates a @tech{nested flow} for a paper abstract.}
```
It generates exactly `\begin{syntax}`.

dvanhorn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+
+Similar to @racket[include-section], but incorporates the document in the
+specified module as an abstract. The document must have no title or
+sub-parts.}
```
This is just like the existing sigplan library.

dvanhorn commented on scribble-doc/scribblings/scribble/acmart.scrbl:
```diff
@@ -0,0 +1,157 @@
+@defproc[(maketitle) block?]{
+
+Issues the @tt{\maketitle} command.  This must appear after the abstract
+any several other top-matter commands.  (See the acmart latex documentation.)}
```
1. Yes.
2. Yes.
3. I think that would be a mess.  The better thing would be to automatically insert `maketitle` at the right point, but I couldn't figure out how to do it.

dvanhorn commented:
I've taken care of all the FIXMEs (but one) and I believe addressed all your comments.  The last remaining issue is what to do about the theorem styles.  Not sure if they should be included in the library or not.

mflatt commented:
When I try this, I get `cannot open input file "acmart-load.tex"`. Is that file missing?

mflatt commented:
Also, doc rendering reports
```
raco setup: WARNING: undefined tag in <pkgs>/scribble-doc/scribblings/scribble/scribble.scrbl:
raco setup:  ((lib "scribble/acmart.rkt") email)
```

dvanhorn commented:
I fixed both of these issues, @mflatt.  Thanks!

mflatt commented:
@dvanhorn I still get the same error, but changing `"acmart-load.tex"` to `(scribble-file "acmart/acmart-load.tex")` now fixes that much.

But then I get `! LaTeX Error: \SConferenceInfo undefined.` for `\renewcommand{\SConferenceInfo}`. Do I need to try something more than
```
#lang scribble/acmart
@title{Example}
```
?

dvanhorn commented:
It looks like you at least need an abstract, but I'm not sure why.  I'll try to fix.

mflatt commented:
It looks like `title` is also needed before `maketitle` – not surprising, but probably worth mentioning in the docs. Also, if there's no good way around having to use `maketitle`, probably `maketitle` should just be mentioned at the beginning of the `scribble/acmart` docs.

dvanhorn commented:
@mflatt, your example now works once `maketitle` is added (I wasn't using the style properties pervasively that triggered the tex-addition).  Will document `maketitle` earlier and more prominently.

dvanhorn commented:
I moved `maketitle` up, included a minimal example, and specified that it is mandatory.
